fstest/fstests/fstests.go

```diff
@@ -196,6 +196,45 @@ func PutTestContentsMetadata(ctx context.Context, t *testing.T, f fs.Fs, file *f
 	return obj
 }
 
+// PutTestContentsOptions is not compatible with the way that the b2 backend handles metadata. This function accepts
+// metadata as options to be passed to Put() rather than metadata to be set in the src ObjectInfo
+func PutTestContentsOptions(ctx context.Context, t *testing.T, f fs.Fs, file *fstest.Item, contents string, check bool, mimeType string, options ...fs.OpenOption) fs.Object {
+	var (
```

Comment:
I think you can probably add `options ...fs.OpenOption` as the last parameter of PutTestContentsMetadata for a neater fix.

This needs refactoring to take an options object, but that can be another change.

Reply:
Will do - in fact, I think I intended to go back and do exactly that!